### ISSUE TITLE
API Gateway Error: HTTP_PROXY GET not yet implemented

### ISSUE THREAD

**Issue by justwrote** [localstack/localstack#1941]
When setting up a API Gateway with HTTP_PROXY I get the following error message when accessing the route `http://localhost:4567/restapis/xaper8hype/test/_user_request_/test123`

```
{"message": "API Gateway integration type \"HTTP_PROXY\" for method \"GET\" not yet implemented"}
```

tf file
```
provider "aws" {
  access_key                  = "mock_access_key"
  region                      = "us-east-1"
  s3_force_path_style         = true
  secret_key                  = "mock_secret_key"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true

  endpoints {
    apigateway     = "http://localhost:4567"
    cloudformation = "http://localhost:4581"
    cloudwatch     = "http://localhost:4582"
    dynamodb       = "http://localhost:4569"
    es             = "http://localhost:4578"
    firehose       = "http://localhost:4573"
    iam            = "http://localhost:4593"
    kinesis        = "http://localhost:4568"
    lambda         = "http://localhost:4574"
    route53        = "http://localhost:4580"
    redshift       = "http://localhost:4577"
    s3             = "http://localhost:4572"
    secretsmanager = "http://localhost:4584"
    ses            = "http://localhost:4579"
    sns            = "http://localhost:4575"
    sqs            = "http://localhost:4576"
    ssm            = "http://localhost:4583"
    stepfunctions  = "http://localhost:4585"
    sts            = "http://localhost:4592"
  }
}

resource "aws_api_gateway_rest_api" "MyDemoAPI" {
  name        = "MyDemoAPI"
  description = "This is my API for demonstration purposes"
}

resource "aws_api_gateway_resource" "MyDemoResource" {
  rest_api_id = aws_api_gateway_rest_api.MyDemoAPI.id
  parent_id   = aws_api_gateway_rest_api.MyDemoAPI.root_resource_id
  path_part   = "test123"
}

resource "aws_api_gateway_method" "MyDemoMethod" {
  rest_api_id   = aws_api_gateway_rest_api.MyDemoAPI.id
  resource_id   = aws_api_gateway_resource.MyDemoResource.id
  http_method   = "GET"
  authorization = "NONE"
   request_parameters = {
    "method.request.path.proxy" = true
  }
}

resource "aws_api_gateway_integration" "MyDemoIntegration" {
  rest_api_id          = aws_api_gateway_rest_api.MyDemoAPI.id
  resource_id          = aws_api_gateway_resource.MyDemoResource.id
  http_method          = aws_api_gateway_method.MyDemoMethod.http_method
  type                 = "HTTP_PROXY"
  uri                  = "https://httpbin.org/anything/test123"
  integration_http_method = "GET"

  cache_key_parameters = ["method.request.path.proxy"]

  timeout_milliseconds = 29000
  request_parameters = {
    "integration.request.path.proxy" = "method.request.path.proxy"
  }
}

resource "aws_api_gateway_deployment" "teststage" {
  depends_on = [
    aws_api_gateway_integration.MyDemoIntegration
  ]
  rest_api_id = aws_api_gateway_rest_api.MyDemoAPI.id
  stage_name  = "test"
}

output "base_url" {
  value = aws_api_gateway_deployment.teststage.invoke_url
}
```

docker compose
```
version: '3'
services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack:latest
    ports:
      - "4567-4597:4567-4597"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=apigateway
      - DEBUG=1
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

aws output
```
$ aws --endpoint=http://localhost:4567 --region us-east-1 apigateway get-resources --rest-api-id xaper8hype
{
    "items": [
        {
            "id": "5t3u62jjyg",
            "path": "/",
            "resourceMethods": {
                "GET": {}
            }
        },
        {
            "id": "fu3fph78jj",
            "parentId": "5t3u62jjyg",
            "pathPart": "test123",
            "path": "/test123",
            "resourceMethods": {
                "GET": {
                    "httpMethod": "GET",
                    "authorizationType": "NONE",
                    "methodIntegration": {
                        "type": "HTTP_PROXY",
                        "httpMethod": "GET",
                        "uri": "https://httpbin.org/anything/test123",
                        "requestTemplates": {},
                        "integrationResponses": {
                            "200": {
                                "statusCode": 200,
                                "responseTemplates": {
                                    "application/json": null
                                }
                            }
                        }
                    }
                }
            }
        }
    ]
}
```

I'm not sure if I do something wrong but the terraform code does work on AWS.

**Comment by n9iels** [comment]
It would be really nice if these kind of limitations are documented somewhere... I spent a well amount of time on this issue without knowing it was a limitation of localstack. My understanding was that the POST method was somehow not implemented for the given Lamda.....

**Comment by whummer** [comment]
Thanks for reporting @justwrote . This should be fixed in #2787 - can you please do a `docker pull localstack/localstack` and give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
